### PR TITLE
WIP: Issue510 usecond

### PR DESCRIPTION
**Pull request by PRemmen**
This is a suggestion to solve a couple of issues, namely:

#583 
#581
#580
#578
#553
#539
#510
#487
#438

### CODE REVIEW

**Comment by PRemmen** [comment]
the unit tests run, however there is a problem with python version 3.4. @MichaMans do you have any idea?

**Comment by MichaMans** [comment]
@PRemmen If I am not mistaken here, pandas does not support python 3.4 . So do we, if we introduce a dependency of pandas. Feasible fix from my point of view. Delete the 3.4 checks and remove it from the list of support python versions 😃 
But with the implemented workaround the tests are passing 😄

**Comment by PRemmen** [comment]
@MichaMans The coverage decreased because of now untested functions that will be deleted in the next version of TEASER (see https://github.com/RWTH-EBC/TEASER/issues/508#issuecomment-487141130) so I think we can discuss if we need to cover them at all

**Comment by PRemmen** [comment]
@PRemmen ToDo: convert dansih tabula files to json!

**Comment by PRemmen** [comment]
@MichaMans push push push :)

**Comment by MichaMans** [comment]
okokok. I'll try to have a look in the days. In the meantime, coverage is going down. As i understand, we are changing the data classes from xml and pyxb to json and "handcoded" imports? Do we provide both in the suggested implementation? Then we should test both and the coverage should not decrease. If we want to release that in a lighter version with no backwards compatibility, then we should delete the classes and related tests. @PRemmen whats your plan?

**Comment by PRemmen** [comment]
@MichaMans I'll have a look into the coverage in more detail at the weekend. However, I think the other implementations are not affected by that

**Comment by MichaMans** [comment]
@PRemmen sure! I already started to have a look at the new implementation and will continue that.

**Comment by PRemmen** [comment]
grafik

This is the coverage statistics, solving coverage in material and type_elements input (old) is easy. Not really sure yet where we tested the 0.3.9. binding? :D but I think this is the main reason why the coverage decreases that much

**Comment by PRemmen** [comment]
actually, this was quite a good hint @MichaMans I accidentally changed a file path of a test, now the coverage is back to 97%.

File: tests/test_useconditions.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                utilities.get_default_path(), 'UseCondUT.json'))
+        except OSError:
+            pass
+        path = os.path.join(
+            utilities.get_default_path(), 'UseCondUT.json')
+        prj.data.path_uc = path
+        prj.data.load_uc_binding()
+        use_cond = prj.buildings[-1].thermal_zones[-1].use_conditions
+        use_cond.save_use_conditions(data_class=prj.data)
+        use_cond.save_use_conditions(data_class=prj.data)
```
[the remark this answers is not in the document]
Comment (PRemmen) [reply tests/test_useconditions.py]: Yes this is a duplicate, I'll dete it.

To you concernc, you are absolutely right. my proposal would be to have a separate issue for that :-) but merge the current status of this branch into dev

[the remark this answers is not in the document]
Comment (PRemmen) [reply tests/test_useconditions.py]: Ah sorry, it is not duplicated, this tests if a usage that already is in json is not saved twice, I'll write a separate test.

[the remark this answers is not in the document]
Comment (MichaMans) [reply tests/test_useconditions.py]: ahh ok, makes sense! Alright! Then i'm willing to approve this if you reopen an issue for this 😃

**Comment by PRemmen** [comment]
If test is running I think we are ready to merge, what do you think? @MichaMans

**Comment by MichaMans** [comment]
@PRemmen you are ready to merge 😃

**Pull request merged**